modules/xfem/src/constraints/XFEMSingleVariableConstraint.C

```diff
@@ -40,6 +45,14 @@ XFEMSingleVariableConstraint::XFEMSingleVariableConstraint(const InputParameters
     _jump_flux(getParam<Real>("jump_flux")),
     _use_penalty(getParam<bool>("use_penalty"))
 {
+  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(_fe_problem.getXFEM());
+  if (_xfem == NULL)
+    mooseError("Problem casting to XFEM in XFEMSingleVariableConstraint");
+
+  const UserObject * uo =
+      &(_fe_problem.getUserObjectBase(getParam<UserObjectName>("geometric_cut_userobject")));
+
+  _interface_id = _xfem->getGeometricCutID(dynamic_cast<const GeometricCutUserObject *>(uo));
```

Comment:
What if this cast fails? If you know this type will always succeed then should just store it by reference and static_cast it. Otherwise, add the check or assertion.

Reply:
check is added